### DISCUSSION TITLE
是否可以提供一种机制方便将go-judge注册到注册中心

### DISCUSSION THREAD

**Discussion by nilIfNull** [criyle/go-judge#61] [Ideas]
目前是fork源码，在`executorserver/main.go`的`initGRPCServer`和`initHTTPServer`中硬编码加入注册和反注册逻辑，但是每次同步源仓库代码时需要解决冲突。

所以，想是不是可以在源仓库中提供一种机制来方便扩展？

> [dubbo-go registry](https://github.com/apache/dubbo-go/tree/main/registry)

**Comment by criyle** [comment]
最近对 http 和 grpc 的逻辑有修改来支持 ipv4 ipv6 双架构网络，之后不太会有对于 http grpc 大改了。

**Reply by nilIfNull** [reply]
嗯，看到了这块的修改。已经合并到二开代码，感谢您！

注册中心支持的特性暂时不考虑支持，还是没其他人提需求就后期不考虑了？

**Reply by criyle** [reply]
个人认为加入来自供应商特定代码是不好拓展而且加大耦合。使用脚本来同时启动和注册服务 或者 使用 side car 容器 或者 使用服务网络是比较解耦合的。

**Reply by nilIfNull** [reply]
确实是这样，会增加非必要耦合。感谢提供建议
